addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1266,6 +1270,86 @@ options.registry.parallax = options.Class.extend({
     onMove: function () {
         this._refreshPublicWidgets();
     },
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * Build/remove parallax.
+     *
+     * @see this.selectClass for parameters
+     */
+    parallax: function (previewMode, widgetValue, params) {
+        const isParallax = widgetValue === 'true';
+        this.$target.toggleClass('parallax', isParallax);
+        if (isParallax) {
+            this._initParallax();
+        } else {
+            this._removeParallax();
+        }
+        this.isParallax = isParallax;
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    _computeVisibility: function () {
+        let $el = this.$target.find('> .s_parallax_bg');
+        $el = $el[0] ? $el : this.$target;
+        const hasBackgroundImage = $el.css('background-image') !== 'none';
+        if (!hasBackgroundImage) {
+            this._removeParallax();
+            this.$target.removeClass('oe_img_bg');
+        }
+        const showParallaxOptions = hasBackgroundImage && !this.$target.hasClass('o_background_video');
+        return this._super(...arguments) && showParallaxOptions;
+    },
+    /**
+     * @private
+     */
+    _initParallax: function () {
+        this._updateBackgroundOptions();
+        // FIXME big hack, this should be improved
+        if (!this.isParallax) {
+            const colorPicker = this._requestUserValueWidgets('bg_color_opt')[0];
+            colorPicker._previewColor = 'black-50';
+            colorPicker._value = 'black-50';
+            colorPicker.notifyValueChange(false);
+        }
```

Comment:
In fact, could not you use the data-trigger-widgets system to do that?

Reply:
The problem is that I set the filter with 'black-50' on one condition: If we put a parallax mode and that we were not in parallax before. So the user keeps his filter if he has already chosen one. If use data-trigger-widgets I would force the user to change his filter to black-50 each time he changes a parallax mode.

Reply:
Ok